### PR TITLE
Separate a few debug stuff from CONFIG_DEBUG_MM

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Separate a few debug stuff from CONFIG_DEBUG_MM.

* backtrace
* dump and abort on failure

## Testing
compile tested a few combinations
tested on esp32 with these new options disabled

### CODE REVIEW

**Review by hartmannathan — APPROVED** [review]
Looks good to me.

**Comment by yamt** [comment]
i pushed a lot of unrelated stuff to the branch mistakenly and then restored.

**Review by pkarashchenko — APPROVED** [review]

File: mm/mm_heap/mm_malloc.c [hunk 1]
```diff
@@ -244,15 +244,18 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
...
 
       mwarn("WARNING: Allocation failed, size %zu\n", alignsize);
+#ifdef CONFIG_MM_DUMP_ON_FAILURE
       mm_mallinfo(heap, &minfo);
       mwarn("Total:%d, used:%d, free:%d, largest:%d, nused:%d, nfree:%d\n",
             minfo.arena, minfo.uordblks, minfo.fordblks,
             minfo.mxordblk, minfo.aordblks, minfo.ordblks);
       mm_memdump(heap, -1);
-      DEBUGASSERT(false);
+#endif
```
Comment (gustavonihei) [inline mm/mm_heap/mm_malloc.c:258]:
```suggestion
      DEBUGPANIC();
#endif
```
Was the removal of the `DEBUGASSERT(false);` intended?
I believe leaving it here may be valuable during debugging.

Comment (gustavonihei) [reply mm/mm_heap/mm_malloc.c:258]: Ok, just noticed the separate commit just for this change. It makes sense.

Comment (xiaoxiang781216) [reply mm/mm_heap/mm_malloc.c:258]: where are the separate commit?

Comment (xiaoxiang781216) [reply mm/mm_heap/mm_malloc.c:258]: but DEBUGASSERT get removed?

Comment (pkarashchenko) [reply mm/mm_heap/mm_malloc.c:258]: Yes. There is an explanation in commit message. Maybe debug assert should be optional, I do not know.

Comment (xiaoxiang781216) [reply mm/mm_heap/mm_malloc.c:258]: DEBUGPANIC/DEBUGASSERT could generate the callstack which is very useful to know the abnormal location, so it's better to keep the as before. If user want to see heap info, he should also want to see the callback info too.

Comment (yamt) [reply mm/mm_heap/mm_malloc.c:258]: ok. i added it back as a separate option.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**